pkg/converter/convert.go

```diff
@@ -119,10 +120,66 @@ func Convert(o Options) error {
 				exec.Args = append(exec.Args, serveridArg)
 				exec.Args = append(exec.Args, argLoginMethod)
 				exec.Args = append(exec.Args, o.TokenOptions.LoginMethod)
-			} else {
-				// others are not supported yet
+
+			case token.ServicePrincipalLogin, token.ROPCLogin, token.DeviceCodeLogin, token.MSILogin, token.WorkloadIdentityLogin:
+				if !isAlternativeLogin && o.isSet(flagEnvironment) {
+					exec.Args = append(exec.Args, argEnvironment)
+					exec.Args = append(exec.Args, o.TokenOptions.Environment)
+				} else if !isAlternativeLogin && authInfo.AuthProvider.Config[cfgEnvironment] != "" {
+					exec.Args = append(exec.Args, argEnvironment)
+					exec.Args = append(exec.Args, authInfo.AuthProvider.Config[cfgEnvironment])
+				}
+				if o.isSet(flagServerID) {
+					exec.Args = append(exec.Args, argServerID)
+					exec.Args = append(exec.Args, o.TokenOptions.ServerID)
+				} else if authInfo.AuthProvider.Config[cfgApiserverID] != "" {
+					exec.Args = append(exec.Args, argServerID)
+					exec.Args = append(exec.Args, authInfo.AuthProvider.Config[cfgApiserverID])
+				}
+				if o.isSet(flagClientID) {
+					exec.Args = append(exec.Args, argClientID)
+					exec.Args = append(exec.Args, o.TokenOptions.ClientID)
+				} else if !isAlternativeLogin && authInfo.AuthProvider.Config[cfgClientID] != "" {
```

Comment:
with exec format, you wont have authProvider. You will need to implement similar function like you did for getServerId() above